### PR TITLE
bugfix(facebook pixel): timestamp check

### PR DESCRIPTION
**Pull request by sauravlal-rudderstack**
## Description of the change

> Timestamp check, for payloads which have been sent within 7 days or up to 1min, will only be processed further else it is discarded with a message.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/facebook_pixel/transform.js [hunk 1]
```diff
@@ -410,6 +411,19 @@ const responseBuilderSimple = (
     MAPPING_CONFIG[CONFIG_CATEGORIES.COMMON.name],
     "fb_pixel"
   );
+  const start = moment.unix(commonData.event_time);
```
[the remark this answers is not in the document]
Comment (sauravlal-rudderstack) [reply v0/destinations/facebook_pixel/transform.js:414]: Actually, the timestamp is a generic field, so in processEvent, we have to put multiple check conditions where the timestamps actually are. After mapping, we can easily check these.

[the remark this answers is not in the document]
Comment (utsabc) [reply v0/destinations/facebook_pixel/transform.js:414]: I dont think that is a good reason to do it after all there computation, it basically waste of compute resources. We are mapping from `originalTimeStamp` || `timeStamp` so checking them in order of presence would do the job

**Comment by utsabc** [comment]
> We should delete the timestamp from test cases otherwise it will always fail

I meant deleting in test.js not inputs
you can use delete operator 
```
 it(`${name} - payload: ${index}`, async () => {
        try {
          delete input.event.originalTimestamp
          const output = await transformer.process(input);
          expect(output).toEqual(expectedData[index]);
        } catch (error) {
          expect(error.message).toEqual(expectedData[index].error);
        }
```

File: v0/destinations/facebook_pixel/transform.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sauravlal-rudderstack) [reply v0/destinations/facebook_pixel/transform.js]: Done

**Comment by utsabc** [comment]
Test cases are still failing

File: __tests__/fbpixel.test.js [hunk 2]
```diff
@@ -30,6 +30,8 @@ describe(`${name} Tests`, () => {
     inputData.forEach((input, index) => {
       it(`${name} - payload: ${index}`, async () => {
         try {
+          delete input.event.originalTimestamp;
+          delete input.event.timestamp;
```
Comment (utsabc) [inline __tests__/fbpixel.test.js:34]:
```suggestion
          delete input.message.originalTimestamp;
          delete input.message.timestamp;
```

**Comment by utsabc** [comment]
Lets remove all the hardcoded time stamps `originalTimestamp` or `timeStamp` from input payload  and dynamically add those in the fbpixel.test.js just before `const output = await transformer.process(input);` also same for router transform tests

**Comment by ItsSudip** [comment]
> Lets remove all the hardcoded time stamps `originalTimestamp` or `timeStamp` from input payload and dynamically add those in the fbpixel.test.js just before `const output = await transformer.process(input);` also same for router transform tests

In this approach you need to update the output testcases as well.

**Comment by utsabc** [comment]
> > Lets remove all the hardcoded time stamps `originalTimestamp` or `timeStamp` from input payload and dynamically add those in the fbpixel.test.js just before `const output = await transformer.process(input);` also same for router transform tests
> 
> In this approach you need to update the output testcases as well.

Yes you need to dynamically update the output test cases

File: __tests__/fbpixel.test.js (hunk 2 above)
Comment (sauravlal-rudderstack) [reply __tests__/fbpixel.test.js]: Updated

**Review by ItsSudip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**